src/lib/support/Span.h

```diff
@@ -46,6 +46,13 @@ class Span
     constexpr explicit Span(T (&databuf)[N]) : Span(databuf, N)
     {}
 
+    template <size_t N>
+    constexpr Span& operator=(T (&databuf)[N]) {
```

Comment:
I don't see this used anywhere in this PR.  Please do a separate PR for this if it's needed for something.  I'm not sure we should be adding this at all.

Reply:
I use this fairly broadly in `TestClusterObjectUtils.cpp` to make it easier to initialize a field within the generated object to point to a statically sized array:

E.g:

```
        chip::app::clusters::TestCluster::SimpleStruct::Type t;
        uint8_t buf[4] = {0, 1, 2, 3};
     
        t.d = buf;
```

Reply:
I may be wrong, but it seems the operator us used in https://github.com/project-chip/connectedhomeip/pull/9500/files#diff-98012ee7404b72ff6b6c9a5e8aee4c97c4fb548ad6202cc4e724a29b347b720aR165

Reply:
Ah, I had missed the bytespan case in there.  OK.